### PR TITLE
allow setting grant_type for device code

### LINKED ISSUE #113: allow setting grant_type for device code
The `grant_type` for device code authentication using Google is `http://oauth.net/grant_type/device/1.0` https://github.com/dermesser/yup-oauth2/blob/0ab7c4926694a8373d10d468ee012081394cdd10/src/device.rs#L351 [as documented](https://developers.google.com/identity/protocols/OAuth2ForDevices#step-4:-poll-googles-authorization-server). However, the `grant_type` required by Azure for device code flow is `urn:ietf:params:oauth:grant-type:device_code`. https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-device-code

### PR DESCRIPTION
**Pull request by ctaggart**
fix #113

### CODE REVIEW

File: src/device.rs [hunk 1]
```diff
@@ -147,6 +163,7 @@ where
         let client = self.client.clone();
         let wait = self.wait;
         let mut fd = self.fd.clone();
+        println!("device_code_url {:?}", self.device_code_url);
```
Comment (ctaggart) [inline src/device.rs]: oops

**Comment by ctaggart** [comment]
A working modified test-device example of setting the grant_type for Azure is:

``` rust
    let device_code_url = format!("https://login.microsoftonline.com/{}/oauth2/v2.0/devicecode", &aad_tenant_id);
    let grant_type = "urn:ietf:params:oauth:grant-type:device_code";
    let creds = yup_oauth2::read_application_secret(path::Path::new("clientsecret.json")).expect("clientsecret");
    let device_flow =
        DeviceFlow::new(creds)
            .device_code_url(device_code_url)
            .grant_type(grant_type.to_owned());
    let mut auth =
        Authenticator::new(device_flow)
            .persist_tokens_to_disk("tokenstorage.json")
            .build()
            .expect("authenticator");

    let scopes = vec!["user.read", "openid", "profile"];
    let mut rt = tokio::runtime::Runtime::new().unwrap();
    let fut = auth.token(scopes).and_then(|tok| Ok(println!("{:?}", tok)));

    println!("{:?}", rt.block_on(fut));
```

**Comment by ctaggart** [comment]
A squash merge is preferred.

**Comment by ggriffiniii** [comment]
FYI: I've incorporated this change into my queue of pending changes in PR #112

**Comment by ctaggart** [comment]
Awesome! I'd prefer that PR be merged instead of this.

**Pull request closed without merging**